components/automate-ui/src/app/entities/layout/layout.facade.ts

```diff
@@ -11,63 +11,111 @@ import { MenuItemGroup } from './layout.model';
 import { sidebarMenuGroups, showPageLoading } from './layout.selectors';
 import { ShowPageLoading, UpdateSidebarMenuGroups } from './layout.actions';
 
+import { GetProjects } from 'app/entities/projects/project.actions';
+
 // Important! These must match components/automate-ui/src/styles/_variables.scss
 enum Height {
-  Navigation = '70px',
-  Banner = '110px'
+  Navigation = 70,
+  License = 39,
+  ProcessProgressBar = 48,
+  PendingEditsBar = 36
 }
 
 @Injectable({
   providedIn: 'root'
 })
-export class LayoutFacadeService {
-  headerHeight = '70px';
-  contentHeight = `calc(100% - ${this.headerHeight})`;
-  menuGroups$: Observable<MenuItemGroup[]>;
-  showPageLoading$: Observable<boolean>;
-  showLicenseNotification = false;
-  showHeader = true;
-  showSidebar = true;
+export class LayoutFacadeService implements OnDestroy {
+  public layout = {
+    header: {
+      display: true,
+      license: false,
+      navigation: true
+    },
+    sidebar: {
+      display: true,
+      navigation: true
+    },
+    userNotifications: {
+      display: true,
+      pendingEdits: false,
+      updatesProcessing: false
+    }
+  };
+  public contentHeight = `calc(100% - ${Height.Navigation}px)`;
+  public menuGroups$: Observable<MenuItemGroup[]>;
+  public showPageLoading$: Observable<boolean>;
+  private isDestroyed = new Subject<boolean>();
 
   constructor(
     private store: Store<fromLayout.LayoutEntityState>,
     private layoutSidebarService: LayoutSidebarService
   ) {
+    this.store.dispatch(new GetProjects());
     this.menuGroups$ = store.select(sidebarMenuGroups);
     this.showPageLoading$ = store.select(showPageLoading);
 
     store.select(notificationState).subscribe(
       (notifications: Notification[]) => {
-        this.showLicenseNotification =
+        this.layout.header.license =
           notifications &&  notifications.some(n => n.type === Type.license);
-        this.updateContentHeight(
-          this.showLicenseNotification ? Height.Banner : Height.Navigation);
+          if (this.layout.header.license) {
+            this.layout.header.license = true;
+            this.updateDisplay();
+          } else {
+            this.layout.header.license = false;
+            this.updateDisplay();
+          }
```

Comment:
Not sure what is going on here but I suspect it was the result of multiple iterations of this block of code. Looks like this entire block can be replaced with this:
```
this.updateDisplay();
```

Reply:
@msorens so this code is needed it's to turn on and off the display of the license notifications but with that said I had a mental note to move it. I've now moved the code to the notifications component 👍🏻